src/internal/internal_he2hb_trmm.cc

```diff
@@ -179,91 +183,65 @@ void he2hb_trmm(
-                int64_t nb0 = B.tileNb( 0 );
-                int64_t mb1 = B.tileMb( B.mt()-1 );
-                int64_t nb1 = B.tileNb( B.mt()-1 );
-
-                rank_lower = -1;
-                rank_upper = -1;
-
-                for (int64_t i = 0; i < i_interior; ++i) {
-                    for (int64_t j : panel_rank_rows) {
-                        if (i >= j) { // lower
-                            rank_lower = AH.tileRank( i, j );
-                        }
-                        else { // upper
-                            rank_upper = AH.tileRank( j, i );
-                        }
-                    }
-                    A0 = A.sub( 0, 0, 0, 0 );
-                    int64_t mb = A0.tileMb( 0 );
-                    int64_t nb = A0.tileNb( 0 );
-                    auto Bi = B.sub( i, i, 0, 0 );
-                    bool trapezoid = (mb < nb);
-
-                    if (trapezoid) {
-                        auto B00 = Bi( 0, 0 );
-                        mb1 = B00.mb();
-                        A0 = A0.slice( 0, mb-1,  0, mb-1 ); // first mb-by-mb part
-                        Bi = Bi.slice( 0, mb1-1, 0, mb-1 ); // first mb1-by-mb part
-                    }
-                    auto T = TriangularMatrix<scalar_t>( Uplo::Upper, Diag::NonUnit, A0 );
-
-                    if (rank_upper == mpi_rank || rank_lower == mpi_rank) {
-                        if (device == B.tileDevice( i, 0 )) {
-                            a_array0.push_back( T( 0, 0, device ).data() );
-                            b_array0.push_back( Bi( 0, 0, device ).data() );
-                            //b_array0.push_back( B( i, 0, device ).data() );
-                            lda0 = A0( 0, 0, device ).stride();
-                            ldb0 = Bi( 0, 0, device ).stride();
-                            mb0 = Bi.tileMb( 0 );
-                            nb0 = Bi.tileNb( 0 );
-                        }
-                    }
+                int64_t mb = A0.tileMb( 0 );
+                int64_t nb = A0.tileNb( 0 );
+                bool trapezoid = (mb < nb);
+                if (trapezoid) {
+                    A0 = A0.slice( 0, mb-1,  0, mb-1 ); // first mb-by-mb part
                 }
+                auto T = TriangularMatrix<scalar_t>( Uplo::Upper, Diag::NonUnit, A0 );
 
-                if (i_last == 1) {
-                    int64_t i = B.mt()-1;
-                    rank_lower = -1;
-                    rank_upper = -1;
-                    for (int64_t j : panel_rank_rows) {
-                        if (i >= j) { // lower
-                            rank_lower = AH.tileRank( i, j );
-                        }
-                        else { // upper
-                            rank_upper = AH.tileRank( j, i );
+                scalar_t** t_array_host = B.array_host(device, queue_index);
+                scalar_t** b_array_host = t_array_host + batch_size;
+
+                // Varient of device_regions_build to handle trsmA
```

Comment:
"trsmA"? This is trmm.
Maybe these should say something like "Variant of device_regions_build; similar (or same) code used in trsmA."
Also spelling of variant.

Reply:
The connection to `trsmA` is just that it was the first function I made a variant for and copied that to the other two.  I'll fix the comments to reference the respective functions.

Variant is one of those words I never spell correctly without spellcheck.